detectable/src/main/java/com/synopsys/integration/detectable/detectables/docker/DockerExtractor.java

```diff
@@ -119,7 +127,7 @@ private void importTars(final List<File> importTars, final File directory, final
         }
     }
 
-    private Extraction executeDocker(final File outputDirectory, final String imageArgument, final String imagePiece, final String dockerTarFilePath, final File directory, final File javaExe, final File bashExe,
+    private Extraction executeDocker(final File outputDirectory, final String imageArgument, String imagePiece, final String dockerTarFilePath, final File directory, final File javaExe, final File bashExe,
```

Comment:
My own personal preference (I wouldn't be surprised if opinions vary on this) would be to rename imagePiece to something like givenImagePiece, or suppliedImagePiece, and keep it as final. Then in the method, figure out what the value for DOCKER_IMAGE_NAME_META_DATA needs to be, and assign that value to a new (also final) variable.

Reply:
I agree that makes more sense